caddyhttp/proxy/upstream_test.go

```diff
@@ -448,3 +450,56 @@ func TestHealthCheckPort(t *testing.T) {
 	})
 
 }
+
+func TestHealthCheckContentString(t *testing.T) {
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		fmt.Fprintf(w, "blablabla good blablabla")
+		r.Body.Close()
+	}))
+	_, port, err := net.SplitHostPort(server.Listener.Addr().String())
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer server.Close()
+
+	tests := []struct {
+		config        string
+		contentString string
+		shouldContain bool
+	}{
+		{"proxy / localhost:8080 {\n health_check \n health_check_port " + port +
+			" \n health_check_contains good\n}",
+			"good",
+			true,
+		},
+		{"proxy / localhost:8080 {\n health_check \n health_check_port " + port +
+			" \n health_check_contains good\n}",
+			"bad",
+			false,
+		},
+	}
+	for i, test := range tests {
+		_, err := NewStaticUpstreams(caddyfile.NewDispenser("Testfile", strings.NewReader(test.config)), "")
```

Comment:
Why we supply a invalid config? I think we should create a valid `*staticUpstream` and call its `healthCheck` for test.